compiler/src/dotty/tools/dotc/transform/init/Semantic.scala

```diff
@@ -124,7 +124,7 @@ class Semantic {
        */
       def updateField(field: Symbol, value: Value): Contextual[Unit] =
         val fields = heap(ref).fields
-        assert(!fields.contains(field), field.show + " already init, new = " + value + ", ref =" + ref)
+        // assert(!fields.contains(field), field.show + " already init, new = " + value + ", ref =" + ref)
```

Comment:
The fix-point evaluation needs to update fields that are already initialized, but this line doesn't allow to do that. Could we remove this assertion, or shall we find other ways to empty the heap before re-evaluation?